validate/error.go

```diff
@@ -0,0 +1,52 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+)
+
+// In the specifications in the above table of contents, the keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+type RFC_LEVEL int
+
+const (
+	RFC_MUST RFC_LEVEL = iota
+	RFC_MUST_NOT
+	RFC_REQUIRED
+	RFC_SHALL
+	RFC_SHALL_NOT
+	RFC_SHOULD
+	RFC_SHOULD_NOT
+	RFC_RECOMMENDED
+	RFC_MAY
+	RFC_OPTIONAL
+)
+
+type ERROR_CODE int
+
+const (
+	DEFAULT_FILESYSTEMS = iota
+)
+
+// OCIError represents a error with OCI reference
+type OCIError struct {
+	Level     RFC_LEVEL
+	Reference string
+	Err       error
+}
+
+const ReferencePrefix = "https://github.com/opencontainers/runtime-spec/blob/master/"
```

Comment:
Instead of floating this with `master`, we should be using the `ociVersion` of the config we're checking against.  This is another reason that we [should only be supporting tagged spec releases][1].

[1]: https://github.com/opencontainers/runtime-tools/pull/349#discussion_r108730611